crates/factor-outbound-pg/tests/factor_test.rs

```diff
@@ -0,0 +1,50 @@
+use anyhow::bail;
+use spin_factor_outbound_networking::OutboundNetworkingFactor;
+use spin_factor_outbound_pg::OutboundPgFactor;
+use spin_factor_variables::{StaticVariables, VariablesFactor};
+use spin_factor_wasi::{DummyFilesMounter, WasiFactor};
+use spin_factors::{anyhow, RuntimeFactors};
+use spin_factors_test::{toml, TestEnvironment};
+use spin_world::v2::postgres::HostConnection;
+
+#[derive(RuntimeFactors)]
+struct TestFactors {
+    wasi: WasiFactor,
+    variables: VariablesFactor,
+    networking: OutboundNetworkingFactor,
+    pg: OutboundPgFactor,
+}
+
+fn test_env() -> TestEnvironment {
+    TestEnvironment::default_manifest_extend(toml! {
+        [component.test-component]
+        source = "does-not-exist.wasm"
+    })
+}
+
+#[tokio::test]
+async fn disallowed_host_fails() -> anyhow::Result<()> {
+    let mut factors = TestFactors {
+        wasi: WasiFactor::new(DummyFilesMounter),
+        variables: VariablesFactor::default(),
+        networking: OutboundNetworkingFactor,
+        pg: OutboundPgFactor,
+    };
+    factors.variables.add_provider_type(StaticVariables)?;
+
+    let env = test_env();
+    let mut state = env.build_instance_state(factors).await?;
+
+    let res = state
+        .pg
+        .open("postgres://postgres.test:5432/test".to_string())
+        .await;
+    let Err(err) = res else {
+        bail!("expected Err, got Ok");
+    };
+    assert!(matches!(err, ErrorConnectionFailed)); // TODO: Fix this assert
+
+    Ok(())
+}
+
+// TODO: Write more tests
```

Comment:
Nit: such a todo isn't really meaningful in my opinion and should just be removed.

Reply:
Oh yeah sorry, I'm not done with this PR so this is a TODO for myself to finish up this PR. I marked it draft, but I guess it wasn't super clear b/c I requested reviews.

Agreed that in the abstract this isn't a helpful PR.